File: lib/features/employee/domain/entities/employee.dart
class Employee {
  final int id;
  final String name;
  final String email;
  final String phone;
  final String? position;
  final String? department;
  final double? salary;
  final String? hireDate;

  Employee({
    required this.id,
    required this.name,
    required this.email,
    required this.phone,
    this.position,
    this.department,
    this.salary,
    this.hireDate,
  });
}

File: backend/app/Models/Employee.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Employee extends Model
{
    protected $fillable = [
        'name',
        'email',
        'phone',
        'position',
        'department',
        'salary',
        'hire_date',
        'company_id',
    ];

    protected $casts = [
        'salary' => 'decimal:2',
        'hire_date' => 'date',
    ];

    public function company(): BelongsTo
    {
        return $this->belongsTo(Company::class);
    }
}
